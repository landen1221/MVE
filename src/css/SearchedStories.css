.SearchedStories {
  width: 60%;
  margin: auto;
}

.MuiBox-root {
  border: 1px solid rgba(63, 63, 63, 0.753);
  border-radius: 10px;
  width: 85%;
  margin-left: 0;
  margin-top: 1px;
  margin-bottom: 15px;
}

.SearchedStories-stories {
  align-items: center;
  align-content: center;
  text-align: center;
  margin: auto;
}

#age-gender {
  font-weight: normal;
}

.SearchedStories h4 {
  text-align: left;
  margin-left: 3em;
  margin-bottom: 0;
}

.SearchedStories p {
  text-align: left;
  margin-left: 2.5em;
  margin-top: 0.5em;
  margin-right: 2.5em;
}

.SearchedStories .stories {
  align-items: center;
  align-content: center;
  text-align: center;
  margin: 0;
}

.SearchedStories h3 {
  text-align: left;
  margin-left: 1.5em;
  margin-bottom: 0.5em;
  font-size: 1em;
  /* font-weight: normal; */
  color: rgba(63, 63, 63, 0.753);
}
