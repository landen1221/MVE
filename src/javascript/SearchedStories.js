import { useEffect, useState } from "react";
import MVEAPI from "../api";
import "../css/SearchedStories.css";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Stories from "./Stories";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    flexWrap: "wrap",
  },
}));

const SearchedStories = ({ vaccines, searchBy, search }) => {
  const classes = useStyles();
  const [storyList, setStoryList] = useState([]);

  useEffect(
    function findStories() {
      async function getStories() {
        let tempList = await MVEAPI.searchStories(searchBy);
        setStoryList(tempList.data.results.stories);
      }
      getStories();
    },
    [searchBy]
  );

  return (
    <div className={classes.root}>
      <div className="SearchedStories">
        <h3>
          Stories Matching{" "}
          <u>{searchBy.length > 0 ? searchBy.split("+").join(" ") : "All"}</u>
        </h3>
        {/* <Grid container>
          <SortBox
            dbName={undefined}
            storyList={storyList}
            setStoryList={setStoryList}
            originalStories={originalStories}
          />
        </Grid> */}
        <Grid item xs={12} className="SearchedStories-stories">
          <Stories storyList={storyList} search={search} vaccines={vaccines} />
        </Grid>
      </div>
    </div>
  );
};

export default SearchedStories;
