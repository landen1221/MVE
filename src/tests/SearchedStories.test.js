import React from "react";
import { render } from "@testing-library/react";
import SearchedStories from "../javascript/SearchedStories";
import { vaccines } from "./globalTestVariables";

// Props: vaccines, searchBy, search
const searchBy = "test";
const search = true;

it("renders without crashing", function () {
  render(
    <SearchedStories vaccines={vaccines} searchBy={searchBy} search={search} />
  );
});
