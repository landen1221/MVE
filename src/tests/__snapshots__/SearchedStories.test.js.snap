// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-3"
  >
    <div
      class="SearchedStories"
    >
      <h3>
        Stories Matching 
        <u>
          test
        </u>
      </h3>
      <div
        class="MuiGrid-root SearchedStories-stories MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="Stories makeStyles-root-4"
        >
          <i
            id="no-stories"
          >
            <br />
            no stories available
          </i>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
